src/diart/models.py

```diff
@@ -22,6 +27,15 @@ def __call__(self) -> nn.Module:
         return pyannote_loader.get_model(self.model_info, self.hf_token)
 
 
+class PyannoteWeSpeakerSpeakerEmbeddingLoader:
```

Comment:
I think this should be implemented directly in `PyannoteLoader` instead of creating a new class.
We could replace the `__call__` method of `PyannoteLoader` like this:

```diff
def __call__(self) -> nn.Module:
-        return pyannote_loader.get_model(self.model_info, self.hf_token)
+       return PretrainedSpeakerEmbedding(self.model_info, use_auth_token=self.hf_token)
```

Reply:
I added a different way of handling this, is it okay too?

Reply:
I commented your new version, let's continue the discussion there